tbtadm/controller.cpp

```diff
@@ -519,7 +521,7 @@ void tbtadm::Controller::approve(const fs::path& dir) try
     authorized << 1;
 
     m_out << "Authorized\n";
-    if (m_sl == 2 && !m_once)
+    if (m_sl == 2 && !m_once && hasKeyFile)
```

Comment:
This allows adding a device to ACL in SL2 mode even if it doesn't support SL2.
Even if we decide it is desired (which I don't think so), it'll not work later when `tbtacl` tries to approve it as there is no key.